src/pages/user/post-settings.tsx

```diff
@@ -0,0 +1,18 @@
+import { useAuth } from "../../shared/hooks/useAuth"
+import { AuthError } from "../error/error-auth-page"
+
+import { NavigationMenu } from "../../features/user/components/navigation-menu"
+
+export const PostSettings = () => {
+    const { auth } = useAuth()
+    return(
+       <>
+         {auth ? (
+          <>
+          <NavigationMenu />
```

Comment:
W PostSettings, Profile i Settings używasz <NavigationMenu /> i robisz warunek czy użytkownik jest zalogowany. Wynieś to do pliku np UserDashboardLayout i jako children przekazuj to co się zmienia